src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3282,6 +3289,32 @@ TransactionStateMachine(WorkerSession *session)
 					break;
 				}
 
+				/* if this is a multi-query task, send the next query */
+				if (placementExecution->queryIndex < task->queryCount)
```

Comment:
We've so far used the queryIndex to keep track of which query we sent when there are multiple queries in a single query string. Now we're using it to iterate through a list. 

I'm wondering if there are any edge cases / issues that could result from using queryIndex in two subtly different ways, e.g. could you have multiple queries in each query string in a list and which would cause queryIndex to overflow? Do we need to prevent that? Would it make sense to introduce a second index, or does that add too much complexity?